shared/src/main/scala/scala/xml/Utility.scala

```diff
@@ -243,6 +243,12 @@ object Utility extends AnyRef with parsing.TokenTests {
       }
     }
 
+  // Checks if node when converted to string is an entity ref
+  def checkNodeForEntityRef(n: Node): Boolean = {
+    val st = n.toString
+    st(0) == '&' && st(st.length - 1) == ';'
```

Comment:
@SethTisue I agree.  I'm hoping we can just get rid of this definition in favor of `isAtom`.  If not, then I was going to suggest using `startsWith` and `endsWith` to avoid the inevitable out of bounds errors here.